src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11056,6 +11072,8 @@ private bool ScanExplicitlyTypedLambda(Precedence precedence)
                     // Advance past the open paren or comma.
                     this.EatToken();
 
+                    _ = ParseAttributeDeclarations();
```

Comment:
this is definitely so weird.  i don't expect you to change this in this pr, but at this point, i think we need to rethink the whole 'scan then parse' approach we take.  Now that scanning is effectively just running the parser to make its determination, we should think about just swithcing to a "just parse" approach.  We'd still have rewind points, we just wouldn't be trying to duplicate the logic between these guys.